pkg/build/controller/strategy/util.go

```diff
@@ -139,6 +142,22 @@ func setupSourceSecrets(pod *kapi.Pod, sourceSecret *kapi.LocalObjectReference)
 	}...)
 }
 
+// setupSecretBuildSources mounts the secrets referenced by the SecretBuildSource
+// into a builder container. It also sets an environment variable that contains
+// a name of the secret and the destination directory.
+func setupSecretBuildSources(pod *kapi.Pod, secrets []buildapi.SecretBuildSource) {
+	for _, s := range secrets {
+		mountSecretVolume(pod, s.Secret.Name, filepath.Join(SecretBuildSourceBaseMountPath, s.Secret.Name), "build")
```

Comment:
@bparres do we want to support creating builds where the secret is the only
input? IOW. is it valid to have Source without Git or ImageSource or
Dockerfile, just with the secret?

On Mon, Jan 18, 2016 at 7:15 PM, Michal Fojtik <mi@mifo.sk> wrote:

> secret names has to be valid DNS characters, I think '/' is not allowed :-)
>
> On Mon, Jan 18, 2016 at 6:59 PM, Ben Parees <notifications@github.com>
> wrote:
>
>> In pkg/build/controller/strategy/util.go
>> <https://github.com/openshift/origin/pull/6636#discussion_r50027775>:
>>
>> > @@ -139,6 +142,22 @@ func setupSourceSecrets(pod *kapi.Pod, sourceSecret *kapi.LocalObjectReference)
>> >  	}...)
>> >  }
>> >
>> > +// setupSecretBuildSources mounts the secrets referenced by the SecretBuildSource
>> > +// into a builder container. It also sets an environment variable that contains
>> > +// a name of the secret and the destination directory.
>> > +func setupSecretBuildSources(pod *kapi.Pod, secrets []buildapi.SecretBuildSource) {
>> > +	for _, s := range secrets {
>> > +		mountSecretVolume(pod, s.Secret.Name, filepath.Join(SecretBuildSourceBaseMountPath, s.Secret.Name), "build")
>>
>> similarly i'd be worried here about allowing users to append arbitrary
>> secret name values as a path. what if my secret is named "../../.."? could
>> something bad happen here?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/6636/files#r50027775>.
>>
>
>
>
> --
> -----------------------------------------------
> Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
> Red Hat OpenShift, Engineering
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering